File: prime.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import random
import math

class PrimeNumber:

    def __init__(self):
        self.primes = [3, 5, 7]
        self.normal = [self.getRand(), self.getRand(), self.getRand()]

    #Random integer between 1-10
    def getRand(self) -> int:
        return random.randrange(1,11)

    #A
    def primeFactor(self) -> int:
        return ((self.primes[0]**self.normal[0]) * (self.primes[1]**self.normal[1]) * (self.primes[2]**self.normal[2]))

    #Positive full divisor
    def fullDivisor(self) -> int:
        return ((self.normal[0]+1) * (self.normal[1]+1) * (self.normal[2]+1))

    #Sum of positive full divisor
    def fullDivisorSum(self) -> int:
        return ( (((self.primes[0]**(self.normal[0]+1))-1)/(self.primes[0]-1)) * (((self.primes[1]**(self.normal[1]+1))-1)/(self.primes[1]-1)) * (((self.primes[2]**(self.normal[2]+1))-1)/(self.primes[2]-1)) )

    #Sum of non-prime integer divisors
    def nonPrimeDivisor(self) -> int:
        return -(sum(self.primes))

    #Coprime integers with A
    def coprime(self) -> int:
        return self.primeFactor() * ((self.primes[0]-1)/self.primes[0]) * ((self.primes[1]-1)/self.primes[1]) * ((self.primes[2]-1)/self.primes[2])

    #Product of positive full divisors of number A
    def product(self) -> float:
        return math.sqrt((self.primeFactor() ** ( (self.normal[0]+1) * (self.normal[1]+1) * (self.normal[2]+1) )))

    #Check if a number is prime
    def isPrime(self, n) -> bool:
        #Corner case
        if (n <= 1):
            return false;

        #Check from 2 to n-1
        for i in range(2,n):
            if (n % i == 0):
                return false

        return true



if __name__ == '__main__':
    obj = PrimeNumber()
    print('Prime factor:', obj.primeFactor())
    print('Full divisor:', obj.fullDivisor())
    print('Sum of full divisor:', obj.fullDivisorSum())
    print('Non-Prime divisor:', obj.nonPrimeDivisor())
    print('Coprime: ', obj.coprime())
    #print(obj.product()) #it's too long to print

File: prime.js
class PrimeNumber {

  constructor() {
    //Prime Numbers (a, b, c)
    this.primes = [3,5,7];

    //Other numbers (d, e, f)
    this.normal = [this.randomInt(1,10),this.randomInt(1,10),this.randomInt(1,10)];
  }

  //A
  primeFactor(){
    return ((this.primes[0]**this.normal[0]) * (this.primes[1]**this.normal[1]) * (this.primes[2]**this.normal[2]));
  }

  //Positive full divisor
  fullDivisor(){
    return ((this.normal[0]+1) * (this.normal[1]+1) * (this.normal[2]+1));
  }

  //Sum of positive full divisor
  fullDivisorSum(){
    return ( (((this.primes[0]**(this.normal[0]+1))-1)/(this.primes[0]-1)) * (((this.primes[1]**(this.normal[1]+1))-1)/(this.primes[1]-1)) * (((this.primes[2]**(this.normal[2]+1))-1)/(this.primes[2]-1)) );
  }

  //Sum of non-prime integer divisors
  nonPrimeDivisor(){
    return -(this.primes.reduce((a, b) => a + b, 0));
  }

  //Coprime integers with A
  coprime(){
    return this.primeFactor() * ((this.primes[0]-1)/this.primes[0]) * ((this.primes[1]-1)/this.primes[1]) * ((this.primes[2]-1)/this.primes[2]);
  }

  //Product of positive full divisors of number A
  product(){
    return Math.sqrt((this.primeFactor() ** ( (this.normal[0]+1) * (this.normal[1]+1) * (this.normal[2]+1) )));
  }

  //Random integer between 1-10
  randomInt(min, max) {
    return Math.floor(Math.random() * (max - min)) + min;
  }

  isPrime(n){
    // Corner case
    if (n <= 1)
        return false;

    // Check from 2 to n-1
    for (let i = 2; i < n; i++)
        if (n % i == 0)
            return false;

    return true;
  }

}

const pm = new PrimeNumber();
console.log("Prime factor: " + pm.primeFactor());
console.log("Full divisor: " + pm.fullDivisor());
console.log("Sum of full divisor: " + pm.fullDivisorSum());
console.log("Non-prime divisor: " + pm.nonPrimeDivisor());
console.log("Coprime: " + pm.coprime());
//console.log(pm.product());  //it's too long to print
